### PR TITLE
test: E2E database connection test for postgres

### PR DESCRIPTION
**Pull request by hessjcg**
Adds an e2e test that actually connects to the postgres database and executes a query.

Related-to: #52

### CODE REVIEW

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.11.1
+    controller-gen.kubebuilder.io/version: v0.11.2
```
Comment (enocom) [inline config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:18]: Is this a separate change?

File: internal/testhelpers/resources.go [hunk 2]
```diff
@@ -49,6 +49,83 @@ func buildPodTemplateSpec(mainPodSleep int) corev1.PodTemplateSpec {
 	}
 }
 
+func BuildSecret(secretName, userKey, user, passwordKey, password, dbNameKey, dbName string) corev1.Secret {
```
Comment (enocom) [inline internal/testhelpers/resources.go:52]: Exported functions should have a docstring, even if it's short and internal to the project.

File: internal/testhelpers/resources.go [hunk 3]
```diff
@@ -271,6 +348,57 @@ func (cc *TestCaseClient) ExpectPodContainerCount(ctx context.Context, podSelect
...
+				countBadPods++
+			}
+		}
+		switch {
+		case allOrAny == "all" && countBadPods > 0:
+			return fmt.Errorf("got %d pods not ready of %d pods, want 0 pods not ready", countBadPods, len(pods.Items))
+		case allOrAny == "any" && countBadPods == countPods:
+			return fmt.Errorf("got  %d pods not ready of %d pods, want at least 1 pod ready ", countBadPods, len(pods.Items))
+		}
+		return nil
```
Comment (enocom) [inline internal/testhelpers/resources.go:382]: This return can move up into a `default` case.

File: internal/testhelpers/resources.go [hunk 4]
```diff
@@ -271,6 +348,57 @@ func (cc *TestCaseClient) ExpectPodContainerCount(ctx context.Context, podSelect
...
+// containers on the deployment's PodSpec.Containers == count. Returns error after 30 seconds
+// if the containers do not match.
+func (cc *TestCaseClient) ExpectPodReady(ctx context.Context, podSelector *metav1.LabelSelector, allOrAny string) error {
+
+	var (
+		countBadPods int
+		countPods    int
+	)
+
+	err := RetryUntilSuccess(24, DefaultRetryInterval, func() error {
```
Comment (enocom) [inline internal/testhelpers/resources.go:361]: This can just be `return RetryUntilSucess( /* ... */ )`.

**Review by enocom — CHANGES_REQUESTED** [review]
A few minor things. Otherwise this looks good.

File: internal/testhelpers/resources.go (hunk 2 above)
Comment (hessjcg) [reply internal/testhelpers/resources.go:52]: Fixed. I added a few more too.

File: internal/testhelpers/resources.go (hunk 3 above)
Comment (hessjcg) [reply internal/testhelpers/resources.go:382]: Fixed.

File: internal/testhelpers/resources.go (hunk 4 above)
Comment (hessjcg) [reply internal/testhelpers/resources.go:361]: Fixed.

**Review by enocom — APPROVED** [review]

**Pull request merged**